web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -250,6 +249,34 @@ public void contextInitialized(ServletContextEvent event) {
 				servletContext.setAttribute(WebApplicationContext.ROOT_WEB_APPLICATION_CONTEXT_ATTRIBUTE, context);
 				
 				WebDaemon.startOpenmrs(event.getServletContext());
+	
+				// new addition: detect port (works with mvn jetty:run or -Djetty.http.port=9999)
+				final String contextPath = servletContext.getContextPath();
+				int port = 8080; // default
+				
+				try {
+					String jettyPort = System.getProperty("jetty.http.port");
```

Comment:
this is for the jetty server which is okay, now we need to also cater for cargo where in order to override the default port, you provide a property through `cargo.servlet.port`

Reply:
You could do something like `String[] portProps = {"jetty.http.port","cargo.servlet.port""};` and use a foreach loop to loop through `portProps`